applications/serial_lte_modem/doc/TCPIP_AT_commands.rst

```diff
@@ -0,0 +1,1541 @@
+
+* The ``<port>`` parameter is an integer and it represents the TCP service port.
+  It is mandatory for starting the server.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSVR: <handle> started
+
+The ``<handle>`` value is an integer.
+When positive, it indicates that it opened successfully.
+When negative, it indicates that it failed to open.
+
+Unsolicited notification
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSVR: <error> stopped
+
+The ``<error>`` value is a negative integer, and it represents the error value according to the standard POSIX errorno.
+
+::
+
+   #XTCPDATA: <datatype>, <size>
+
+* The ``<datatype>`` value can assume one the following values:
+
+  * ``0`` - hexidecimal string (e.g. "DEADBEEF" for 0xDEADBEEF)
+  * ``1`` - plain text (default value)
+  * ``2`` - JSON
+  * ``3`` - HTML
+  * ``4`` - OMA TLV
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xtcpsvr=1,3442
+   #XTCPSVR: 2 started
+   OK
+   #XTCPSVR: 5.189.130.26 connected
+   #XTCPRECV: 1, 13
+   Hello, TCP#1!
+   #XTCPRECV: 1, 13
+   Hello, TCP#2!
+
+Read command
+------------
+
+The read command allows you to check the TCP server settings.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPSVR?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSVR: <listen_socket_handle>,<income_socket_handle>,<data_mode>
+
+The ``<handle>`` value is an integer.
+When positive, it indicates that it opened successfully.
+When negative, it indicates that it failed to open or that there is no incoming connection.
+
+* The ``<data_mode>`` value can assume one the following values:
+
+  * ``0`` - Disabled
+  * ``1`` - Enabled
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xtcpsvr?
+   #XTCPSVR: 1, 2, 0
+   OK
+   #XTCPSVR: timeout
+   at#xtcpsvr?
+   #XTCPSVR: 1, -1
+   OK
+
+Test command
+------------
+
+The test command tests the existence of the command and provides information about the type of its subparameters.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPSVR=?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSVR: (list of op value),<port>,<sec_tag>
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xtcpsvr=?
+   #XTCPSVR: (0, 1, 2),<port>,<sec_tag>
+   OK
+
+TCP/TLS client #XTCPCLI
+=======================
+
+The ``#XTCPCLI`` command allows you to create a TCP/TLS client and to connect to a server.
+
+Set command
+-----------
+
+The set command allows you to create a TCP/TLS client and to connect to a server.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPCLI=<op>[,<url>,<port>[,[sec_tag]]
+
+* The ``<op>`` parameter can accept one the following values:
+
+  * ``0`` - Disconnect
+  * ``1`` - Connect to the server
+  * ``2`` - Connect to the server with data mode support
+
+* The ``<url>`` parameter is a string and it indicates the hostname or the IP address to connect to.
+  Its maximum size is 128 bytes.
+  When the parameter is an IP address, it supports IPv4 only, not IPv6.
+* The ``<port>`` parameter is an integer and it represents the TCP/TLS service port.
+  It is mandatory for starting the server.
```

Comment:
More details on the Sec_tag are needed here too

Reply:
explained in above